openedx_learning/apps/core/itemstore/models.py

```diff
@@ -0,0 +1,133 @@
+"""
+The item model hiearachy is: Item -> ItemVersion -> ItemInfo -> ItemRaw
+
+Item is the versionless thing that is guaranteed to exist for the lifetime of
+the LearningContext. An ItemVersion is a different version of that item for a
+given LearningContext, and may include policy changes (like grading). ItemInfo
+covers basic metadata that is intrinsic to the item itself, and now how it's
+used in a LearningContext. ItemRaw represents the raw byte data.
+
+TODO: Add link to ADR after it merges.
+"""
+from django.db import models
+
+from openedx_learning.lib.fields import (
+    hash_field,
+    identifier_field,
+    immutable_uuid_field,
+    manual_date_time_field,
+)
+from ..publish.models import LearningContext, LearningContextVersion
+
+
+class Item(models.Model):
+    """
+    This represents any content that has ever existed in a LearningContext.
+    """
+    uuid = immutable_uuid_field()
+    learning_context = models.ForeignKey(LearningContext, on_delete=models.CASCADE)
+    identifier = identifier_field()
+
+    # TODO: These pointers to the latest published version are convenient, but
+    # we don't currently have the data integrity guarantees to make sure that 
+    # multiple versions aren't active at the same time. Maybe add a ref to the
+    # Item model in LearningContextVersionItemVersion, so we can force that
+    # constraint?
+    published_version = models.ForeignKey('ItemVersion', on_delete=models.RESTRICT, null=True, related_name="+")
+    first_published_at = models.DateTimeField(null=True)  # can be null if never published 
+    last_published_at = models.DateTimeField(null=True)   # can be null if never published
+
+    created = manual_date_time_field()
+    modified = manual_date_time_field()
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                fields=["learning_context_id", "identifier"],
```

Comment:
Yeah. I think that in general, the `identifier` fields have to be namespaced so that it's entirely local to one particular LearningContext.

Reply:
In the case of XBlocks, do you see this `identifier` as being the usage key (`block-v1:a+b+c+type@video+block@xyz`) or the block id (just `xyz`)?

Reply:
I'd prefer not to have the full learning context key encoded, but since we currently allow blocks of different types to have the same ID, I think we have to encode at least something like `video+xyz`